Assets/Scripts/Interactables/LaunchArcRenderer.cs

```diff
@@ -65,7 +65,12 @@ Vector3 CalculateArcPoint(float t, float maxDistance)
     private void OnEnable()
     {
         // Allows arc to be rendered in correct position when re-enabled
-        Update();
+        if (hasBeenDisabled) Update();
+    }
+
+    private void OnDisable()
+    {
+        hasBeenDisabled = true;
```

Comment:
I'm confused about what this is doing... if there's some edge case you're trying to cover here, can we write a comment about it?
Also do we need to reset this bool at any point back to false, or is this just a one time thing?

Reply:
It's just a one-time thing, I can leave a comment about it.

Basically, I want to call Update() from OnEnable(), but the behavior I really want is to only call Update() when the GameObject is RE-enabled (so not between Awake() and Start() like OnEnable() normally does).

Do you know of any good workaround for this?